tests/func/test_storage_commands.py

```diff
@@ -0,0 +1,200 @@
+        "team": team or "team_name",
+        "team_name": team or "team_name",
+    }
+
+
+def test_cp_storage_local_to_local(studio_token, tmp_dir):
+    (tmp_dir / "path1").mkdir(parents=True, exist_ok=True)
+    (tmp_dir / "path1" / "file1.txt").write_text("file1")
+    (tmp_dir / "path2").mkdir(parents=True, exist_ok=True)
+
+    result = main(
+        [
+            "cp",
+            "-s",
+            str(tmp_dir / "path1" / "file1.txt"),
+            str(tmp_dir / "path2" / "file1.txt"),
+        ]
+    )
+    assert result == 0
+
+    assert (tmp_dir / "path2" / "file1.txt").read_text() == "file1"
+
+
+def test_cp_storage_local_to_s3(requests_mock, capsys, studio_token, tmp_dir):
+    (tmp_dir / "path1").mkdir(parents=True, exist_ok=True)
+    (tmp_dir / "path1" / "file1.txt").write_text("file1")
+
+    requests_mock.post(
+        f"{STUDIO_URL}/api/datachain/storages/batch-presigned-urls",
+        json={
+            "urls": {
+                "data/content": {
+                    "url": "https://example.com/upload",
+                    "fields": {"key": "value"},
+                }
+            },
+            "headers": {},
+            "method": "POST",
+        },
+    )
+    requests_mock.post("https://example.com/upload", status_code=200)
+    requests_mock.post(
+        f"{STUDIO_URL}/api/datachain/storages/activity-logs",
+        json={"success": True},
+    )
+
+    result = main(
+        [
+            "cp",
+            "-s",
+            str(tmp_dir / "path1" / "file1.txt"),
+            "s3://my-bucket/data/content",
+        ]
+    )
+    assert result == 0
+
+    history = requests_mock.request_history
+    assert len(history) == 3
+    assert history[0].url == f"{STUDIO_URL}/api/datachain/storages/batch-presigned-urls"
+    assert history[1].url == "https://example.com/upload"
+    assert history[2].url == f"{STUDIO_URL}/api/datachain/storages/activity-logs"
+
+    assert history[0].json() == {
+        "bucket": "my-bucket",
+        "paths": {"data/content": "text/plain"},
+        "remote": "s3",
+        "team": "team_name",
+        "team_name": "team_name",
+    }
+
+
+def test_cp_remote_to_local(requests_mock, capsys, studio_token, tmp_dir):
+    requests_mock.get(
+        f"{STUDIO_URL}/api/datachain/storages/files/download?bucket=my-bucket&remote=s3&filepath=data%2Fcontent&team=team_name&team_name=team_name",
+        json={
+            "url": "https://example.com/download",
+        },
+    )
+    requests_mock.get(
+        "https://example.com/download",
+        content=b"file1",
+    )
+
+    result = main(
+        ["cp", "-s", "s3://my-bucket/data/content", str(tmp_dir / "file1.txt")]
+    )
+    assert result == 0
+    assert (tmp_dir / "file1.txt").read_text() == "file1"
+
+    history = requests_mock.request_history
+    assert len(history) == 2
+    assert history[1].url == "https://example.com/download"
+
+
+def test_cp_s3_to_s3(requests_mock, capsys, studio_token, tmp_dir):
```

Comment:
it is probably not a functional test

Reply:
(and review other tests here)

Reply:
Why not? From perspective of datachain, it starts with the cli invocation (main) and reaches to studio (third party).